### ISSUE TITLE
Order by multiple columns

### ISSUE THREAD

**Issue by mhier** [saltcorn/saltcorn#1757]
**Is your feature request related to a problem? Please describe.**
I would like to order a list view by multiple columns, e.g. the member list of an orchestra shall be sorted first by instrument and second by name.

**Describe the solution you'd like**
In the configuration options for a list view, currently only one default sorting field can be selected (and the direction). It should be possible to add multiple fields there, with each a direction. Ideally, it should also be possible somehow for the user showing the list view to select multiple columns for sorting.

**Comment by j2l** [comment]
To sort, you could add a tabulator view and click on column sort (triangle) in the order.

**Comment by mhier** [comment]
Thanks for the hint. I didn't know of the tabulator view module before, but I like it very much. It even allows to sort for joined fields which does not work in lists (cf. #1176). Unfortunately what you suggested does not work. Clicking on the column sort seems to clear any previous sort requests.

I don't see how filter views would help. I want to sort, not hide the other entries. Can you clarify what you meant exactly?

**Comment by j2l** [comment]
Yes, filters was just a way for you to get the result, but it hides data.

I don't think tabulator has settings to set it view only. But if your user has only read privilege, and not write, modification shouldn't pass through.
The other solution is to fork the tabulator plugin on github and tweak it.

**Comment by itachiemashita89** [comment]
Hello..
Sorry if I disturb this conversation.
I do test this Tabulator View, but seem having error "cannot POST/View/
Cannot find solution to make view appear.
Appreacite if any clue to fix this.
Thanks.

![Screenshot 2023-04-27 153544](https://user-images.githubusercontent.com/91925990/234792596-74ab182e-4960-4d01-b45e-5253c256728d.png)

**Comment by j2l** [comment]
@itachiemashita89 which table is it? Users?
Double check the table rights and view rights (user, staff, admin).

**Comment by itachiemashita89** [comment]
Hello @j2l ,
Thank you for your respond.
Yes table Users. View right already set as admin and login also as admin.
Set to lower user also cannot get view.
Only able get the button view but not showing the table.

![Screenshot 2023-05-02 090658](https://user-images.githubusercontent.com/91925990/235558098-d9ee7374-e825-4460-9b25-bbb581946cf0.png)

**Comment by itachiemashita89** [comment]
Hello,
Update. I think got bug before.
I do uninstall modules Tabulator..restart server..then re-install back.
Then do add view for Tabulator on the selected table, able to get view.
Thanks.
Really nice. I really need this Tabulator for my simple system.
Thank you Thank you and Thank you...
![Screenshot 2023-05-02 104530](https://user-images.githubusercontent.com/91925990/235568801-cb0df8a6-f643-4ed1-aaf9-4054d625b583.png)

**Comment by j2l** [comment]
You're welcome @itachiemashita89 !

**Comment by mhier** [comment]
> EDIT: Ah, the tabulator view is editable. Is there any way to set it to view only?

I found out you can configure for each column whether it is editable or not. Nice!

Still, I am missing the originally requested feature of sorting by multiple columns. It would be really great if one could sort a tabular view as described by @j2l (clicking on several column headers sequentially in reverse order), as well as configuring multiple default sorting columns to define the sort order when opening the view.

**Comment by itachiemashita89** [comment]
Hello @mhier ,

Just sharing back.
My version currently use is 0.8.6-beta.2
For disable can tick under column configuration. Once tick, for sure the row cannot edit.

![disable editing](https://user-images.githubusercontent.com/91925990/235611624-67caddf1-a787-463f-9165-9db0228bca50.png)

And for Header filter, I just enable the function here. Then all record able search individually.

![Filter](https://user-images.githubusercontent.com/91925990/235611693-3693611c-bb48-4744-9fb7-c31a8444f13e.png)

Hope this able assist you as well.
Thanks.

**Comment by j2l** [comment]
Thanks for sharing @itachiemashita89 !

**Comment by mhier** [comment]
> Hope this able assist you as well.

Yes, that's what I already found out on my own. Thanks anyway :-) I am still missing the possibility to sort by multiple columns, as described above.

**Comment by SpameaterAtAnbraDotDe** [comment]
I would also like to be able to sort the data by several columns.
